diagnostics/mixed_layer_depth/mixed_layer_depth.py

```diff
@@ -0,0 +1,285 @@
+    
+    Returns
+    -------
+    mld: xarray.DataArray, dims of time, space
+
+    This function developed by Dhruv Balweda, Andrew Pauling, Sarah Ragen, Lettie Roach
+
+    """
+    pressure = xr.apply_ufunc(gsw.p_from_z, -fieldthetao.lev, fieldthetao.latitude, output_dtypes=[float,]).rename('pressure')
+    # absolute salinity from practical salinity
+    abs_salinity = xr.apply_ufunc(gsw.SA_from_SP, fieldso, pressure, fieldso.longitude, fieldso.latitude, output_dtypes=[float,]).rename('abs_salinity')
+
+    #calculate cthetao - conservative temperature - from potential temperature
+    cthetao = xr.apply_ufunc(gsw.CT_from_pt, abs_salinity, fieldthetao, output_dtypes=[float,]).rename('cthetao')
+
+    # calculate sigma0 - potential density anomaly with reference pressure of 0 dbar, 
+
+    # this being this particular potential density minus 1000 kg/m^3.
+    sigma0 = xr.apply_ufunc(gsw.density.sigma0, abs_salinity, cthetao, output_dtypes=[float, ]).rename('sigma0')
+
+    # interpolate density data to 10m
+    surf_dens = sigma0.interp(lev=10)
+
+    # density difference between surface and whole field
+    dens_diff = sigma0 - surf_dens
+
+    # keep density differences exceeding threshold, discard other values
+    dens_diff = dens_diff.where(dens_diff > 0.03)
+
+    # level of smallest difference between (density difference to 10m) and (threshold)
+    mld = dens_diff.lev.where(dens_diff==dens_diff.min(['lev'])).max(['lev']).rename('mld')
+
+    # calculate sigma2 - potential density anomaly with reference pressure of 2000 dbar, 
+    # this being this particular potential density minus 1000 kg/m^3.
+    sigma2 = xr.apply_ufunc(gsw.density.sigma2, abs_salinity.isel(time=0), cthetao.isel(time=0), output_dtypes=[float, ]).rename('sigma2') #sigma2.attrs['units']='kg/m^3'
+
+    # compute water depth    
+    test = sigma0.isel(time=0) + sigma0.lev
+    bottom_depth = sigma2.lev.where(test == test.max(dim='lev')).max(dim='lev').rename('bottom_depth') # units 'meters'
+    
+    # set MLD to water depth where MLD is NaN
+    mld = mld.where(~cmath.isnan(mld), bottom_depth)
+
+    return mld
+
+
+
+def computemean(field=None, firstyr=2010, lastyr=2014):
+    """Compute mean
+
+    Parameters
+    ----------
+    field : xarray.DataArray, dims must be time, space (space can be multidim)
+    
+    Returns
+    -------
+    themean, thestd, trend, detrendedstd: xarray.DataArray, dims of month, space
+    residuals: xarray.DataArray, dims of year, month, space
+    """
+    firstyr=int(firstyr)
+    lastyr=int(lastyr)
+    field=xr_reshape(field,'time',['year','month'],[np.arange(firstyr,lastyr+1),np.arange(12)])
+
+    themean = field.mean(dim='year')
+
+    return themean
+
+
+
+### 1) Loading model data files: ###############################################
+
+input_file_so = "{DATADIR}/mon/{CASENAME}.{so_var}.mon.nc".format(**os.environ)
+input_file_thetao = "{DATADIR}/mon/{CASENAME}.{thetao_var}.mon.nc".format(**os.environ)
```

Comment:
These definitions should be removed: absolute paths to all requested variables are set via the `path_variable` setting in the new settings file format, which defaults to `<variable name>_FILE`; see [documentation](https://mdtf-diagnostics.readthedocs.io/en/latest/sphinx/ref_settings.html#varlist-entry-properties). Calls to open_dataset(), etc. below should also be changed.